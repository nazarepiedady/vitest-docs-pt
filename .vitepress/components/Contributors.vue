<script setup lang="ts">
import { contributors } from '../contributors'
</script>

<template>
  <div flex="~ wrap gap2" justify-center>
    <a v-for="{ name, avatar } of contributors" :key="name" :href="`https://github.com/${name}`" m-0 rel="noopener noreferrer" :aria-label="`${name} na GitHub`">
      <img loading="lazy" :src="avatar" width="50" height="50" rounded-full h-12 w-12 :alt="`avatar do ${name}`">
    </a>
  </div>
</template>
