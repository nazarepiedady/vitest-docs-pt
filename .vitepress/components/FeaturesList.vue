<template>
  <ul
    class="features-list"
    dir="auto"
    flex="~ col gap2 md:gap-3"
  >
    <ListItem>
      Configuração, transformadores, resolvedores, e extensões da <a target="_blank" href="https://vitejs.dev" rel="noopener noreferrer">Vite</a>.
    </ListItem>
    <ListItem>
      Use a mesma configuração da tua aplicação para executar os testes!
    </ListItem>
    <ListItem>
      <a
        target="_blank"
        href="https://twitter.com/antfu7/status/1468233216939245579"
        rel="noopener noreferrer">
        Modo de Observação Inteligente e Instantânea, como HMR para testes!
      </a>
    </ListItem>
    <ListItem>
      Testagem de Componente para Vue, React, Svelte, Lit e Muito Mais.
    </ListItem>
    <ListItem>
      Suporte de TypeScript / JSX Fora da Caixa
    </ListItem>
    <ListItem>
      Módulo de ECMAScript em primeiro lugar, espera de alto nível
    </ListItem>
    <ListItem>
      Operários com várias linhas de processamento através da <a target="_blank" href="https://github.com/tinylibs/tinypool" rel="noopener noreferrer">Tinypool</a>
    </ListItem>
    <ListItem>
      Suporte a analise comparativa com a <a target="_blank" href="https://github.com/tinylibs/tinybench" rel="noopener noreferrer">Tinybench</a>
    </ListItem>
    <ListItem>
      Filtragem, pausas, execução simultânea para grupo e testes
    </ListItem>
    <ListItem>
      <a href="/guide/snapshot">
        Fotografia Compatível com a Jest
      </a>
    </ListItem>
    <ListItem>
      <a
        target="_blank"
        href="https://www.chaijs.com/"
        rel="noopener noreferrer">Chai</a> embutida para asserções + as APIs compatíveis com a
        <a
        target="_blank"
        href="https://jestjs.io/docs/expect"
        rel="noopener noreferrer">expect da Jest</a>
    </ListItem>
    <ListItem>
      <a
        target="_blank"
        href="https://github.com/tinylibs/tinyspy"
        rel="noopener noreferrer">Tinyspy</a> embutida para simulação.
    </ListItem>
    <ListItem>
      <a
        target="_blank"
        href="https://github.com/capricorn86/happy-dom"
        rel="noopener noreferrer">happy-dom</a> ou
      <a
        target="_blank"
        href="https://github.com/jsdom/jsdom"
        rel="noopener noreferrer">jsdom</a> para simulação do DOM
    </ListItem>
    <ListItem>
      Cobertura de código através da
      <a
        target="_blank"
        href="https://github.com/bcoe/c8"
        rel="noopener noreferrer">c8</a> ou
      <a
        target="_blank"
        href="https://istanbul.js.org/"
        rel="noopener noreferrer">istanbul</a>
    </ListItem>
    <ListItem>
      <a href="/guide/in-source">Testagem Na Fonte</a> Parecida com a da Rust
    </ListItem>
    <ListItem>
      Testagem de Tipo através da <a target="_blank" href="https://github.com/mmkal/expect-type" rel="noopener noreferrer">expect-type</a>
    </ListItem>
  </ul>
</template>

<style scoped>
.features-list li {
  list-style: none;
  display: flex;
  gap: 0.4rem;
  margin: 0;
}

.features-list {
  padding: 0;
}
</style>
