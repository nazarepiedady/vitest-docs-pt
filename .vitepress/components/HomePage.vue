<script setup lang="ts">
import { VPTeamMembers } from 'vitepress/theme'
import { teamMembers } from '../contributors'
</script>

<template>
  <div class="content">
    <div class="content-container">
      <main class="main">
        <div class="vp-doc" flex flex-col items-center mt-10>
          <h2 id="meet-the-team" op50 font-normal p="t-10 b-2">
            Conhe√ßa A Equipa
          </h2>
          <div w-full p-10>
            <VPTeamMembers size="small" :members="teamMembers" />
          </div>
          <h2 id="the-team" op50 font-normal pt-5 pb-2>
            Colaboradores
          </h2>
          <p text-lg max-w-200 text-center leading-7>
            <Contributors />
            <br>
            <a href="https://chat.vitest.dev" rel="noopener noreferrer">Junta-te a comunidade</a> e envolva-te!
          </p>

          <h2 id="sponsored-by" op50 font-normal pt-5 pb-2>
            Patrocinado por
          </h2>
          <div grid="~ cols-1 lg:cols-2" w-full text-center mt-10>
            <div class="flex flex-col">
              <a text-lg h="32px" href="https://github.com/sponsors/antfu" rel="noopener noreferrer">Patrocinadores do Anthony</a>
              <a href="https://cdn.jsdelivr.net/gh/antfu/static/sponsors.svg" target="_blank" rel="noopener noreferrer">
                <img
                  crossorigin="anonymous"
                  inline-block
                  width="768" height="1464"
                  class="resizable-img"
                  loading="lazy"
                  src="/sponsors/antfu.svg"
                  alt="Patrocinadores do Anthony"
                >
              </a>
            </div>
            <div class="flex flex-col">
              <a text-lg h="32px" href="https://github.com/sponsors/sheremet-va" rel="noopener noreferrer">Patrocinadores do Vladimir</a>
              <a href="https://cdn.jsdelivr.net/gh/sheremet-va/static/sponsors.svg" target="_blank" rel="noopener noreferrer">
                <img
                  crossorigin="anonymous"
                  inline-block
                  width="768" height="1464"
                  class="resizable-img"
                  loading="lazy"
                  src="/sponsors/sheremet-va.svg"
                  alt="Patrocinadores do Vladimir"
                >
              </a>
            </div>
            <div class="flex flex-col">
              <a text-lg h="32px" href="https://github.com/sponsors/patak-dev" rel="noopener noreferrer">Patrocinadores do Patak</a>
              <a href="https://cdn.jsdelivr.net/gh/patak-dev/static/sponsors.svg" target="_blank" rel="noopener noreferrer">
                <img
                  crossorigin="anonymous"
                  inline-block
                  width="768" height="722"
                  class="resizable-img"
                  loading="lazy"
                  src="/sponsors/patak-dev.svg"
                  alt="Patrocinadores do Patak"
                >
              </a>
            </div>
          </div>
          <p class="text-center opacity-75">
            <a href="https://www.netlify.com" rel="noopener noreferrer">
              <img src="/netlify.svg" alt="Servido pela Netlify" width="114" height="151">
            </a>
          </p>
        </div>
      </main>
    </div>
  </div>
</template>
