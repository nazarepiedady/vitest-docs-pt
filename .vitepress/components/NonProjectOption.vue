<template>
  <span
    text-sm
    text-orange
    cursor-not-allowed
    title="Not supported in workspace project config"
  >
    *
  </span>
</template>
