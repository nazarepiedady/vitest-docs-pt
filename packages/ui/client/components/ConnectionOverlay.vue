<script setup lang="ts">
import { client, isConnected, isConnecting } from '~/composables/client'
</script>

<template>
  <template v-if="!isConnected">
    <div
      fixed inset-0 p2 z-10
      select-none
      text="center sm"
      bg="overlay"
      backdrop-blur-sm
      backdrop-saturate-0
      @click="client.reconnect"
    >
      <div
        h-full

        flex="~ col gap-2"
        items-center justify-center
        :class="isConnecting ? 'animate-pulse' : ''"
      >
        <div
          text="5xl"
          :class="isConnecting ? 'i-carbon:renew animate-spin animate-reverse' : 'i-carbon-wifi-off'"
        />
        <div text-2xl>
          {{ isConnecting ? 'Connecting...' : 'Disconnected' }}
        </div>
        <div text-lg op50>
          Check your terminal or start a new server with `vitest --ui`
        </div>
      </div>
    </div>
  </template>
</template>
