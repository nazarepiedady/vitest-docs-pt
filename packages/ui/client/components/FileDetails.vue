<script setup lang="ts">
import { client, current, currentLogs, isReport } from '~/composables/client'
import type { Params } from '~/composables/params'
import { viewMode } from '~/composables/params'
import type { ModuleGraph } from '~/composables/module-graph'
import { getModuleGraph } from '~/composables/module-graph'
import type { ModuleGraphData } from '#types'

const data = ref<ModuleGraphData>({ externalized: [], graph: {}, inlined: [] })
const graph = ref<ModuleGraph>({ nodes: [], links: [] })
const draft = ref(false)
const hasGraphBeenDisplayed = ref(false)

debouncedWatch(
  current,
  async (c, o) => {
    if (c && c.filepath !== o?.filepath) {
      data.value = await client.rpc.getModuleGraph(c.filepath)
      graph.value = getModuleGraph(data.value, c.filepath)
    }
  },
  { debounce: 100, immediate: true },
)

const open = () => {
  const filePath = current.value?.filepath
  if (filePath)
    fetch(`/__open-in-editor?file=${encodeURIComponent(filePath)}`)
}

const changeViewMode = (view: Params['view']) => {
  if (view === 'graph')
    hasGraphBeenDisplayed.value = true

  viewMode.value = view
}
const consoleCount = computed(() => {
  return currentLogs.value?.reduce((s, { size }) => s + size, 0) ?? 0
})
function onDraft(value: boolean) {
  draft.value = value
}
</script>

<template>
  <div v-if="current" flex flex-col h-full max-h-full overflow-hidden>
    <div>
      <div p="2" h-10 flex="~ gap-2" items-center bg-header border="b base">
        <StatusIcon :task="current" />
        <div flex-1 font-light op-50 ws-nowrap truncate text-sm>
          {{ current?.filepath }}
        </div>
        <div class="flex text-lg">
          <IconButton
            v-if="!isReport"
            v-tooltip.bottom="'Open in editor'"
            title="Open in editor"
            icon="i-carbon-launch"
            :disabled="!current?.filepath"
            @click="open"
          />
        </div>
      </div>
      <div flex="~" items-center bg-header border="b-2 base" text-sm h-41px>
        <button
          tab-button
          :class="{ 'tab-button-active': viewMode == null }"
          @click="changeViewMode(null)"
        >
          Report
        </button>
        <button
          tab-button
          :class="{ 'tab-button-active': viewMode === 'graph' }"
          @click="changeViewMode('graph')"
        >
          Module Graph
        </button>
        <button
          v-if="!isReport"
          tab-button
          :class="{ 'tab-button-active': viewMode === 'editor' }"
          @click="changeViewMode('editor')"
        >
          {{ draft ? '*&#160;' : '' }}Code
        </button>
        <button
          tab-button
          :class="{ 'tab-button-active': viewMode === 'console', 'op20': viewMode !== 'console' && consoleCount === 0 }"
          @click="changeViewMode('console')"
        >
          Console ({{ consoleCount }})
        </button>
      </div>
    </div>

    <div flex flex-col flex-1 overflow="hidden">
      <div v-if="hasGraphBeenDisplayed" flex-1>
        <ViewModuleGraph v-show="viewMode === 'graph'" :graph="graph" />
      </div>
      <ViewEditor v-if="viewMode === 'editor'" :key="current.filepath" :file="current" @draft="onDraft" />
      <ViewConsoleOutput v-else-if="viewMode === 'console'" :file="current" />
      <ViewReport v-else-if="!viewMode" :file="current" />
    </div>
  </div>
</template>
