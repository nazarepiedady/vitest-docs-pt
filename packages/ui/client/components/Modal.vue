<script setup lang='ts'>
const props = withDefaults(defineProps<{
  modelValue?: boolean
  direction?: string
}>(), {
  modelValue: false,
  direction: 'bottom',
})

defineEmits<{
  (e: 'update:modelValue', value: boolean): void
}>()

const positionClass = computed(() => {
  switch (props.direction) {
    case 'bottom':
      return 'bottom-0 left-0 right-0 border-t'
    case 'top':
      return 'top-0 left-0 right-0 border-b'
    case 'left':
      return 'bottom-0 left-0 top-0 border-r'
    case 'right':
      return 'bottom-0 top-0 right-0 border-l'
    default:
      return ''
  }
})

const transform = computed(() => {
  switch (props.direction) {
    case 'bottom':
      return 'translateY(100%)'
    case 'top':
      return 'translateY(-100%)'
    case 'left':
      return 'translateX(-100%)'
    case 'right':
      return 'translateX(100%)'
    default:
      return ''
  }
})
</script>

<template>
  <div
    class="fixed inset-0 z-40"
    :class="modelValue ? '' : 'pointer-events-none'"
  >
    <div
      class="bg-base inset-0 absolute transition-opacity duration-500 ease-out"
      :class="modelValue ? 'opacity-50' : 'opacity-0'"
      @click="$emit('update:modelValue', false)"
    />
    <div
      class="bg-base border-base absolute transition-all duration-200 ease-out scrolls"
      :class="[positionClass]"
      :style="modelValue ? {} : { transform }"
    >
      <slot />
    </div>
  </div>
</template>
