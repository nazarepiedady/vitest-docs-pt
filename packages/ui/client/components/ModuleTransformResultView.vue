<script setup lang="ts">
import { client } from '~/composables/client'
const props = defineProps<{ id: string }>()
const emit = defineEmits<{ (e: 'close'): void }>()

const result = asyncComputed(() => client.rpc.getTransformResult(props.id))
const ext = computed(() => props.id?.split(/\./g).pop() || 'js')

const source = computed(() => result.value?.source?.trim() || '')
const code = computed(() => result.value?.code?.replace(/\/\/# sourceMappingURL=.*\n/, '').trim() || '')
const sourceMap = computed(() => ({
  mappings: result.value?.map?.mappings ?? '',
  version: result.value?.map?.version,
}))

onKeyStroke('Escape', () => {
  emit('close')
})
// TODO: sourcemap https://evanw.github.io/source-map-visualization/
</script>

<template>
  <div w-350 max-w-screen h-full flex flex-col>
    <div p-4 relative>
      <p>Module Info</p>
      <p op50 font-mono text-sm>
        {{ id }}
      </p>
      <IconButton icon="i-carbon-close" absolute top-5px right-5px text-2xl @click="emit('close')" />
    </div>
    <div v-if="!result" p-5>
      No transform result found for this module.
    </div>
    <template v-else>
      <div grid="~ cols-2 rows-[min-content_auto]" overflow-hidden flex-auto>
        <div p="x3 y-1" bg-overlay border="base b t r">
          Source
        </div>
        <div p="x3 y-1" bg-overlay border="base b t">
          Transformed
        </div>
        <CodeMirror h-full :model-value="source" read-only v-bind="{ lineNumbers: true }" :mode="ext" />
        <CodeMirror h-full :model-value="code" read-only v-bind="{ lineNumbers: true }" :mode="ext" />
      </div>
      <div v-if="sourceMap.mappings !== ''">
        <div p="x3 y-1" bg-overlay border="base b t">
          Source map (v{{ sourceMap.version }})
        </div>
        <CodeMirror
          :model-value="sourceMap.mappings"
          read-only
          v-bind="{ lineNumbers: true }"
          :mode="ext"
        />
      </div>
    </template>
  </div>
</template>
