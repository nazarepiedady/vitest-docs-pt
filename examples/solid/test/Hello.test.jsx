import { describe, expect, test } from 'vitest'
import { fireEvent, render } from 'solid-testing-library'
import { Hello } from '../components/Hello'

describe('<Hello />', () => {
  test('renders', () => {
    const { container, unmount } = render(() => <Hello count={4} />)
    expect(container.innerHTML).toMatchSnapshot()
    unmount()
  })

  test('updates', async () => {
    const { container, unmount, queryByText } = render(() => <Hello count={4} />)
    const button = queryByText('x1')
    const buttonClicked = new Promise((resolve) => {
      const handler = (ev) => {
        button.removeEventListener('click', handler)
        resolve(ev)
      }
      button.addEventListener('click', handler)
    })
    fireEvent.click(button)
    await buttonClicked
    expect(container.innerHTML).toMatchSnapshot()
    unmount()
  })
})
