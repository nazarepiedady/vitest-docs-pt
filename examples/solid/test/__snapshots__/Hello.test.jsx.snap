// Vitest Snapshot v1

exports[`<Hello /> > renders 1`] = `"<div>4 x 2 = 8</div><button>x1</button>"`;

exports[`<Hello /> > updates 1`] = `"<div>4 x 3 = 12</div><button>x1</button>"`;
